profile_library/shelly/Shelly Plus 2PM/model.json

```
@@ -0,0 +1,29 @@
+{
+  "measure_description": "The measured device fw: 1.4.4, Eco Mode enabled, Wifi Client enabled, WiFi AP disabled, BlueTooth disabled. The result represents the average value of measurements collected from the measurement device API for the following scenarios: dumb device/baseline consumption (~4.9W), all relays turned off, one relay turned on",
+  "measure_method": "manual",
+  "measure_device": "Shelly PM Mini gen3",
+  "measure_device_firmware": "20241011-114503/1.4.4-g6d2a586",
+  "measure_settings": {
+    "SAMPLE_COUNT": 200,
+    "SLEEP_TIME": 0.5
+  },
+  "name": "Shelly Plus 2PM",
+  "aliases": [
+    "SNSW-102P16EU"
+  ],
+  "discovery_by": "entity",
```

```suggestion
  "discovery_by": "device",
```

This must be device, otherwise it will be discovered multiple times, because of multiple switches.
Also see https://docs.powercalc.nl/library/device-types/smart-switch/#smart-switch-with-multiple-relays
Here is the correct and tested example for a multi switch profile.

---

> 1. the same situation you can find in case of Shelly 2.5. You dint complain there

Missed that, but same applies there.

> 2. I've tested that yesterday. There was only one discovered item. It offered both entities assigned to the PC entry during discovery.

Ah I see why it is working.
There is some specific logic for `multi_switch` here: https://github.com/bramstroker/homeassistant-powercalc/blob/master/custom_components/powercalc/helpers.py#L66.
I have added that before we had discovery_by property. Would be nice to just use `discovery_by: device` as I can clean up this specific check in the future and simplify things.
Also it's very clear from the profile that the discovery is done by device.

> 3. will it offer assignment of two entities if I chose discovered by device?

Yes that should work fine with the current profile you have, and also with the suggested change.

---

I have performed the requested changes.
Confirmed with most recent beta (.11)